Dockerfile.icq

```diff
@@ -1,24 +1,21 @@
-FROM golang:1.17-bullseye
+FROM golang:1.17-alpine3.15 AS icq-builder
 
-RUN apt install git
+RUN apk add --upgrade git
 
 WORKDIR /src/app
-RUN git clone https://github.com/Stride-Labs/interchain-queries.git
-RUN cp -r interchain-queries/* .
-RUN rm -rf interchain-queries
-
-# copy over stride module and install it
-RUN mkdir stride_ref
-RUN mkdir stride_ref/stride
-COPY . /src/app/stride_ref/stride
-WORKDIR /src/app/stride_ref/stride
-WORKDIR /src/app
-RUN go mod download
-RUN go build
 
-RUN ln -s /src/app/interchain-queries /usr/local/bin
-RUN adduser --system --home /icq --disabled-password --disabled-login icq -U 1000
-USER icq
+RUN git clone https://github.com/Stride-Labs/interchain-queries.git \
+    && cp -r interchain-queries/* . \
+    && rm -rf interchain-queries \
+    && go build
+
+FROM alpine:3.15
 
+COPY --from=icq-builder /src/app/interchain-queries /usr/local/bin/
+
+RUN adduser -S -h /home/icq -D icq -u 1000
+USER icq
 
+WORKDIR /home/icq
 
+CMD ["interchain-queries", "run"]
```

Comment:
How are hot restarts going to be handled here? Local mode has a while loop, cates has auto restart (afaiu). Is there a similar mechanism here, or could we add one? Not super important, let's only do it if it's a <10 min task

Reply:
This bad boy! https://github.com/Stride-Labs/stride/blob/docker-net-v2/docker-compose.yml#L75